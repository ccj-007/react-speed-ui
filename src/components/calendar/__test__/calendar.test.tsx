import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Calendar } from '../../index';

describe('test Calendar component', () => {
  it('should render base Calendar', () => {
    const { asFragment } = render(<Calendar></Calendar>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('callback is successfully after click event', () => {
    const handleSelect = jest.fn();
    const handlePanelChange = jest.fn();
    render(<Calendar onSelect={handleSelect} onPanelChange={handlePanelChange}></Calendar>);
    fireEvent.click(screen.getByText('15'));
    fireEvent.click(screen.getByText('>'));
    fireEvent.click(screen.getByText('<'));
    expect(handlePanelChange).toHaveBeenCalledTimes(2);
    expect(handleSelect).toHaveBeenCalledTimes(2);
  });

  it('should find classnames', () => {
    const { container } = render(<Calendar></Calendar>);
    expect(container.querySelector('.speed-calendar')).toBeTruthy();
    expect(container.querySelector('.speed-calendar-picker')).toBeTruthy();
    expect(container.querySelector('.speed-calendar-arrow')).toBeTruthy();
    expect(container.querySelector('.speed-calendar-detail')).toBeTruthy();
    expect(container.querySelector('.speed-calendar-week')).toBeTruthy();
    expect(container.querySelector('.speed-calendar-day')).toBeTruthy();
  });

  it('custom data in the Calendar component', () => {
    const Demo = () => {
      let date = new Date();
      const customData = [
        {
          day: 1,
          month: date.getMonth() + 1,
          year: date.getFullYear(),
          getNode: () => {
            return (
              <div style={{ display: 'flex' }}>
                <div>吃饭</div>
                <div
                  style={{
                    background: 'red',
                    width: '5px',
                    height: '5px',
                    borderRadius: '50%',
                    position: 'relative',
                    left: '5px',
                  }}
                ></div>
              </div>
            );
          },
        },
        {
          day: 2,
          month: date.getMonth() + 1,
          year: date.getFullYear(),
          getNode: () => {
            return <div>睡觉</div>;
          },
        },
      ];
      return <Calendar customData={customData}></Calendar>;
    };

    const { asFragment } = render(<Demo></Demo>);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('吃饭')).toBeInTheDocument();
    expect(screen.getByText('睡觉')).toBeInTheDocument();
  });
});
