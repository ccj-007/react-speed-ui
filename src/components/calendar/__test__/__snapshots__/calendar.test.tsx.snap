// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Calendar component custom data in the Calendar component 1`] = `
<DocumentFragment>
  <div
    class="speed-calendar"
  >
    <div
      class="speed-calendar"
    >
      <div
        class="speed-calendar-picker"
      >
        <div
          class="speed-calendar-arrow"
        >
           &lt; 
        </div>
        <div
          class="speed-calendar-detail"
        >
          2022年10月
        </div>
        <div
          class="speed-calendar-arrow"
        >
           &gt; 
        </div>
      </div>
      <div
        class="speed-calendar-week"
      >
        <div
          class="speed-calendar-week-item"
        >
          一
        </div>
        <div
          class="speed-calendar-week-item"
        >
          二
        </div>
        <div
          class="speed-calendar-week-item"
        >
          三
        </div>
        <div
          class="speed-calendar-week-item"
        >
          四
        </div>
        <div
          class="speed-calendar-week-item"
        >
          五
        </div>
        <div
          class="speed-calendar-week-item"
        >
          六
        </div>
        <div
          class="speed-calendar-week-item"
        >
          日
        </div>
      </div>
      <div
        class="speed-calendar-day"
      >
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          27
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          28
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          29
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          30
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          31
        </div>
        <div
          class="speed-calendar-day-item"
        >
          1
          <div
            style="display: flex;"
          >
            <div>
              吃饭
            </div>
            <div
              style="background: red; width: 5px; height: 5px; border-radius: 50%; position: relative; left: 5px;"
            />
          </div>
        </div>
        <div
          class="speed-calendar-day-item"
        >
          2
          <div
            style=""
          >
            睡觉
          </div>
        </div>
        <div
          class="speed-calendar-day-item"
        >
          3
        </div>
        <div
          class="speed-calendar-day-item"
        >
          4
        </div>
        <div
          class="speed-calendar-day-item"
        >
          5
        </div>
        <div
          class="speed-calendar-day-item"
        >
          6
        </div>
        <div
          class="speed-calendar-day-item"
        >
          7
        </div>
        <div
          class="speed-calendar-day-item"
        >
          8
        </div>
        <div
          class="speed-calendar-day-item"
        >
          9
        </div>
        <div
          class="speed-calendar-day-item"
        >
          10
        </div>
        <div
          class="speed-calendar-day-item"
        >
          11
        </div>
        <div
          class="speed-calendar-day-item"
        >
          12
        </div>
        <div
          class="speed-calendar-day-item"
        >
          13
        </div>
        <div
          class="speed-calendar-day-item"
        >
          14
        </div>
        <div
          class="speed-calendar-day-item"
        >
          15
        </div>
        <div
          class="speed-calendar-day-item"
        >
          16
        </div>
        <div
          class="speed-calendar-day-item"
        >
          17
        </div>
        <div
          class="speed-calendar-day-item speed-calendar-day-item-active"
        >
          18
        </div>
        <div
          class="speed-calendar-day-item"
        >
          19
        </div>
        <div
          class="speed-calendar-day-item"
        >
          20
        </div>
        <div
          class="speed-calendar-day-item"
        >
          21
        </div>
        <div
          class="speed-calendar-day-item"
        >
          22
        </div>
        <div
          class="speed-calendar-day-item"
        >
          23
        </div>
        <div
          class="speed-calendar-day-item"
        >
          24
        </div>
        <div
          class="speed-calendar-day-item"
        >
          25
        </div>
        <div
          class="speed-calendar-day-item"
        >
          26
        </div>
        <div
          class="speed-calendar-day-item"
        >
          27
        </div>
        <div
          class="speed-calendar-day-item"
        >
          28
        </div>
        <div
          class="speed-calendar-day-item"
        >
          29
        </div>
        <div
          class="speed-calendar-day-item"
        >
          30
        </div>
        <div
          class="speed-calendar-day-item"
        >
          31
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          1
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          2
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          3
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          4
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          5
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          6
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`test Calendar component should render base Calendar 1`] = `
<DocumentFragment>
  <div
    class="speed-calendar"
  >
    <div
      class="speed-calendar"
    >
      <div
        class="speed-calendar-picker"
      >
        <div
          class="speed-calendar-arrow"
        >
           &lt; 
        </div>
        <div
          class="speed-calendar-detail"
        >
          2022年10月
        </div>
        <div
          class="speed-calendar-arrow"
        >
           &gt; 
        </div>
      </div>
      <div
        class="speed-calendar-week"
      >
        <div
          class="speed-calendar-week-item"
        >
          一
        </div>
        <div
          class="speed-calendar-week-item"
        >
          二
        </div>
        <div
          class="speed-calendar-week-item"
        >
          三
        </div>
        <div
          class="speed-calendar-week-item"
        >
          四
        </div>
        <div
          class="speed-calendar-week-item"
        >
          五
        </div>
        <div
          class="speed-calendar-week-item"
        >
          六
        </div>
        <div
          class="speed-calendar-week-item"
        >
          日
        </div>
      </div>
      <div
        class="speed-calendar-day"
      >
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          27
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          28
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          29
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          30
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          31
        </div>
        <div
          class="speed-calendar-day-item"
        >
          1
        </div>
        <div
          class="speed-calendar-day-item"
        >
          2
        </div>
        <div
          class="speed-calendar-day-item"
        >
          3
        </div>
        <div
          class="speed-calendar-day-item"
        >
          4
        </div>
        <div
          class="speed-calendar-day-item"
        >
          5
        </div>
        <div
          class="speed-calendar-day-item"
        >
          6
        </div>
        <div
          class="speed-calendar-day-item"
        >
          7
        </div>
        <div
          class="speed-calendar-day-item"
        >
          8
        </div>
        <div
          class="speed-calendar-day-item"
        >
          9
        </div>
        <div
          class="speed-calendar-day-item"
        >
          10
        </div>
        <div
          class="speed-calendar-day-item"
        >
          11
        </div>
        <div
          class="speed-calendar-day-item"
        >
          12
        </div>
        <div
          class="speed-calendar-day-item"
        >
          13
        </div>
        <div
          class="speed-calendar-day-item"
        >
          14
        </div>
        <div
          class="speed-calendar-day-item"
        >
          15
        </div>
        <div
          class="speed-calendar-day-item"
        >
          16
        </div>
        <div
          class="speed-calendar-day-item"
        >
          17
        </div>
        <div
          class="speed-calendar-day-item speed-calendar-day-item-active"
        >
          18
        </div>
        <div
          class="speed-calendar-day-item"
        >
          19
        </div>
        <div
          class="speed-calendar-day-item"
        >
          20
        </div>
        <div
          class="speed-calendar-day-item"
        >
          21
        </div>
        <div
          class="speed-calendar-day-item"
        >
          22
        </div>
        <div
          class="speed-calendar-day-item"
        >
          23
        </div>
        <div
          class="speed-calendar-day-item"
        >
          24
        </div>
        <div
          class="speed-calendar-day-item"
        >
          25
        </div>
        <div
          class="speed-calendar-day-item"
        >
          26
        </div>
        <div
          class="speed-calendar-day-item"
        >
          27
        </div>
        <div
          class="speed-calendar-day-item"
        >
          28
        </div>
        <div
          class="speed-calendar-day-item"
        >
          29
        </div>
        <div
          class="speed-calendar-day-item"
        >
          30
        </div>
        <div
          class="speed-calendar-day-item"
        >
          31
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          1
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          2
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          3
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          4
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          5
        </div>
        <div
          class="speed-calendar-day-disabled"
          style="color: rgb(170, 170, 170);"
        >
          6
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
